audio/filter/af_scaletempo2_internals.c

```diff
@@ -480,10 +480,19 @@ static bool can_perform_wsola(struct mp_scaletempo2 *p, double playback_rate)
     return frames_needed(p, playback_rate) <= 0;
 }
 
-static void resize_input_buffer(struct mp_scaletempo2 *p, int size)
+static void grow_input_buffer(struct mp_scaletempo2 *p, int size)
 {
-    p->input_buffer_size = size;
+    assert(size > p->input_buffer_size);
+
     p->input_buffer = realloc_2d(p->input_buffer, p->channels, size);
+
+    float *data = (float*) (p->input_buffer + p->channels);
+    for (int k = p->channels - 1; k > 0; k--) {
+        memmove(data + k * size, data + k * p->input_buffer_size,
```

Comment:
This is a job of `realloc` function. Given that `p->channels` is not big, I don't see any reason to not allocated each channel separately. Specifically `MP_TARRAY_GROW` can be used. There shouldn't be some NIHed memory management in `af_scaletempo2_internals.c` especially that it is broken, doesn't check for allocation failures, doesn't move memory correctly and not only input buffer is affected.

Reply:
Thanks for pointing me into that direction. I gave this a try in #13973, which might make this PR obsolete.